app/src/main/java/com/example/househomey/filter/ui/TagFilterFragment.java

```diff
@@ -1,22 +1,145 @@
+/**
+ * {@code TagFilterFragment} is a subclass of {@link FilterFragment} that provides filtering
+ * functionality based on tags using {@link TagFilter}.
+ */
 package com.example.househomey.filter.ui;
 
+import android.content.Context;
 import android.view.View;
-
+import androidx.core.content.ContextCompat;
+import com.example.househomey.Item;
+import com.example.househomey.R;
 import com.example.househomey.filter.model.FilterCallback;
+import com.example.househomey.filter.model.TagFilter;
+import com.google.android.material.chip.Chip;
+import com.google.android.material.chip.ChipGroup;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
 
+/**
+ * This class represents a fragment for tag-based filtering. It extends {@link FilterFragment}
+ * and implements methods to manage and interact with tag filters.
+ * @author Jared Drueco
+ */
 public class TagFilterFragment extends FilterFragment {
+    protected TagFilter tagFilter;
+    private final ChipGroup tagChipGroup;
+    private Map<String, Boolean> tagSelectionMap = new HashMap<>();
+
+    /**
+     * Constructs a new {@code TagFilterFragment} with the given parameters.
+     *
+     * @param title           The title of the fragment.
+     * @param contentView     The content view of the fragment.
+     * @param filterCallback  The callback for filter actions.
+     * @param itemList        The list of items to be filtered.
+     */
+    public TagFilterFragment(String title, View contentView, FilterCallback filterCallback, ArrayList<Item> itemList) {
+        super(title, contentView, filterCallback);
+        tagChipGroup = contentView.findViewById(R.id.chip_group_labels);
+        populateTagChipGroup(itemList);
+    }
 
-    public TagFilterFragment(String title, View contentView, FilterCallback filterCallback) {
+    /**
+     * Constructs a new {@code TagFilterFragment} with the given parameters, including a
+     * pre-existing {@link TagFilter}.
+     *
+     * @param title           The title of the fragment.
+     * @param contentView     The content view of the fragment.
+     * @param filterCallback  The callback for filter actions.
+     * @param itemList        The list of items to be filtered.
+     * @param tagFilter       The pre-existing tag filter.
+     */
+    public TagFilterFragment(String title, View contentView, FilterCallback filterCallback, ArrayList<Item> itemList, TagFilter tagFilter) {
         super(title, contentView, filterCallback);
+        this.tagFilter = tagFilter;
+        tagSelectionMap = tagFilter.tagSelectionMap;
+        tagChipGroup = contentView.findViewById(R.id.chip_group_labels);
+        tagChipGroup.removeAllViews();
+        populateTagChipGroup(itemList);
```

Comment:
This should all be in onCreateDialog() we are trying to move away from constructors